File: find-the-mine/__tests__/findTheMine.test.js
const { mineLocation } = require('../findTheMine');

describe('Given a 2D array, return the location of the mine (1) in an array representing the mines index [rowIndex, columnIndex]', () => {
    test('Confirm the input array is not mutated', () => {
        const array = [ [1, 0, 0], [0, 0, 0], [0, 0, 0] ];
        mineLocation(array);
        expect(array).toEqual([ [1, 0, 0], [0, 0, 0], [0, 0, 0] ]);
    });
    test('Given [ [1, 0, 0], [0, 0, 0], [0, 0, 0] ], return [0, 0]', () => {
        const array = [ [1, 0, 0], [0, 0, 0], [0, 0, 0] ];
        expect(mineLocation(array)).toEqual([0, 0]);
    });
    test('Given [ [0, 0, 0], [0, 1, 0], [0, 0, 0] ], return [1, 1]', () => {
        const array = [ [0, 0, 0], [0, 1, 0], [0, 0, 0] ];
        expect(mineLocation(array)).toEqual([1, 1]);
    })
    test('Given [ [0, 0, 0], [0, 0, 0], [0, 1, 0] ] , return [2, 1]', () => {
        const array = [ [0, 0, 0], [0, 0, 0], [0, 1, 0] ];
        expect(mineLocation(array)).toEqual([2, 1]);
    });
    test(`Given [
        [
            0, 0, 0, 0, 0,
            0, 0, 0, 0, 0,
            0
        ],
        [
            0, 0, 0, 0, 0,
            0, 0, 0, 0, 0,
            0
        ],
        [
            0, 0, 0, 0, 0,
            0, 0, 0, 0, 0,
            0
        ],
        [
            0, 0, 0, 0, 0,
            0, 0, 0, 0, 0,
            0
        ],
        [
            0, 0, 0, 0, 0,
            0, 0, 0, 0, 0,
            0
        ],
        [
            0, 0, 0, 0, 0,
            0, 0, 0, 0, 0,
            0
        ],
        [
            1, 0, 0, 0, 0,
            0, 0, 0, 0, 0,
            0
        ],
        [
            0, 0, 0, 0, 0,
            0, 0, 0, 0, 0,
            0
        ],
        [
            0, 0, 0, 0, 0,
            0, 0, 0, 0, 0,
            0
        ],
        [
            0, 0, 0, 0, 0,
            0, 0, 0, 0, 0,
            0
        ],
        [
            0, 0, 0, 0, 0,
            0, 0, 0, 0, 0,
            0
        ]], return [6, 0]`, () => {
            
            const array = [
            [
                0, 0, 0, 0, 0,
                0, 0, 0, 0, 0,
                0
            ],
            [
                0, 0, 0, 0, 0,
                0, 0, 0, 0, 0,
                0
            ],
            [
                0, 0, 0, 0, 0,
                0, 0, 0, 0, 0,
                0
            ],
            [
                0, 0, 0, 0, 0,
                0, 0, 0, 0, 0,
                0
            ],
            [
                0, 0, 0, 0, 0,
                0, 0, 0, 0, 0,
                0
            ],
            [
                0, 0, 0, 0, 0,
                0, 0, 0, 0, 0,
                0
            ],
            [
                1, 0, 0, 0, 0,
                0, 0, 0, 0, 0,
                0
            ],
            [
                0, 0, 0, 0, 0,
                0, 0, 0, 0, 0,
                0
            ],
            [
                0, 0, 0, 0, 0,
                0, 0, 0, 0, 0,
                0
            ],
            [
                0, 0, 0, 0, 0,
                0, 0, 0, 0, 0,
                0
            ],
            [
                0, 0, 0, 0, 0,
                0, 0, 0, 0, 0,
                0
            ]
            ]
            expect(mineLocation(array)).toEqual([6, 0]);
        });
});

//https://www.codewars.com/kata/528d9adf0e03778b9e00067e/train/javascript

File: python-venv/find_the_mine/__tests__/find_the_mine_test.py
from kata_function.find_the_mine import mine_location

def test_1():
    array = [ [1, 0, 0], [0, 0, 0], [0, 0, 0] ]
    mine_location(array)
    assert array == [ [1, 0, 0], [0, 0, 0], [0, 0, 0] ]

def test_2():
    array = [ [1, 0, 0], [0, 0, 0], [0, 0, 0] ]
    assert mine_location(array) == [0, 0]

def test_3():
    array = [ [0, 0, 0], [0, 1, 0], [0, 0, 0] ]
    assert mine_location(array) == [1, 1]

def test_4():
    array = [ [0, 0, 0], [0, 0, 0], [0, 1, 0] ]
    assert mine_location(array) == [2, 1]

def test_5():
    array = [
            [
                0, 0, 0, 0, 0,
                0, 0, 0, 0, 0,
                0
            ],
            [
                0, 0, 0, 0, 0,
                0, 0, 0, 0, 0,
                0
            ],
            [
                0, 0, 0, 0, 0,
                0, 0, 0, 0, 0,
                0
            ],
            [
                0, 0, 0, 0, 0,
                0, 0, 0, 0, 0,
                0
            ],
            [
                0, 0, 0, 0, 0,
                0, 0, 0, 0, 0,
                0
            ],
            [
                0, 0, 0, 0, 0,
                0, 0, 0, 0, 0,
                0
            ],
            [
                1, 0, 0, 0, 0,
                0, 0, 0, 0, 0,
                0
            ],
            [
                0, 0, 0, 0, 0,
                0, 0, 0, 0, 0,
                0
            ],
            [
                0, 0, 0, 0, 0,
                0, 0, 0, 0, 0,
                0
            ],
            [
                0, 0, 0, 0, 0,
                0, 0, 0, 0, 0,
                0
            ],
            [
                0, 0, 0, 0, 0,
                0, 0, 0, 0, 0,
                0
            ]
            ]
    assert mine_location(array) == [6, 0]

#https://www.codewars.com/kata/528d9adf0e03778b9e00067e/train/python